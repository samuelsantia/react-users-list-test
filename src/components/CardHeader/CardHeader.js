import styled from 'styled-components/macro';
import PropTypes from 'prop-types';
import { switchBgc } from '../../themes/utils';

/** header background possible colors */
export const bgColors = {
  purple: 'var(--color-purple)',
  red: 'var(--color-red)',
  orange: 'var(--color-orange)',
  yellow: 'var(--color-yellow)',
  green: 'var(--color-green)',
  lightBlue: 'var(--color-light-blue)',
  darkBlue: 'var(--color-dark-blue)',
};

/** renders header for a card */
const CardHeader = styled.header`
  ${switchBgc(bgColors)};
  display: flex;
  justify-content: var(--card-header-h-align, center);
  align-items: var(--card-header-v-align, center);
  padding: var(--card-header-padding, 1rem);
`;

CardHeader.propTypes = {
  /** background color */
  bg: PropTypes.oneOf(Object.keys(bgColors)),
};

export default CardHeader;
