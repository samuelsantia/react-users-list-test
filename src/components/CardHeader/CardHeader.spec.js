import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import CardHeader, { bgColors } from './CardHeader';

describe('<CardHeader />', () => {
  it('it should render', () => {
    const { container } = render(<CardHeader>Test CardHeader</CardHeader>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it.each(Object.keys(bgColors))('should render bgc color %s', (color) => {
    const { container } = render(<CardHeader bgc={color} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
