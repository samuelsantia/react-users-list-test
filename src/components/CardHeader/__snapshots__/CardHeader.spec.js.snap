// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardHeader /> it should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: var(--card-header-h-align,center);
  -webkit-justify-content: var(--card-header-h-align,center);
  -ms-flex-pack: var(--card-header-h-align,center);
  justify-content: var(--card-header-h-align,center);
  -webkit-align-items: var(--card-header-v-align,center);
  -webkit-box-align: var(--card-header-v-align,center);
  -ms-flex-align: var(--card-header-v-align,center);
  align-items: var(--card-header-v-align,center);
  padding: var(--card-header-padding,1rem);
}

<header
  class="c0"
>
  Test CardHeader
</header>
`;

exports[`<CardHeader /> should render bgc color darkBlue 1`] = `
.c0 {
  background-color: var(--color-dark-blue);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: var(--card-header-h-align,center);
  -webkit-justify-content: var(--card-header-h-align,center);
  -ms-flex-pack: var(--card-header-h-align,center);
  justify-content: var(--card-header-h-align,center);
  -webkit-align-items: var(--card-header-v-align,center);
  -webkit-box-align: var(--card-header-v-align,center);
  -ms-flex-align: var(--card-header-v-align,center);
  align-items: var(--card-header-v-align,center);
  padding: var(--card-header-padding,1rem);
}

<header
  class="c0"
/>
`;

exports[`<CardHeader /> should render bgc color green 1`] = `
.c0 {
  background-color: var(--color-green);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: var(--card-header-h-align,center);
  -webkit-justify-content: var(--card-header-h-align,center);
  -ms-flex-pack: var(--card-header-h-align,center);
  justify-content: var(--card-header-h-align,center);
  -webkit-align-items: var(--card-header-v-align,center);
  -webkit-box-align: var(--card-header-v-align,center);
  -ms-flex-align: var(--card-header-v-align,center);
  align-items: var(--card-header-v-align,center);
  padding: var(--card-header-padding,1rem);
}

<header
  class="c0"
/>
`;

exports[`<CardHeader /> should render bgc color lightBlue 1`] = `
.c0 {
  background-color: var(--color-light-blue);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: var(--card-header-h-align,center);
  -webkit-justify-content: var(--card-header-h-align,center);
  -ms-flex-pack: var(--card-header-h-align,center);
  justify-content: var(--card-header-h-align,center);
  -webkit-align-items: var(--card-header-v-align,center);
  -webkit-box-align: var(--card-header-v-align,center);
  -ms-flex-align: var(--card-header-v-align,center);
  align-items: var(--card-header-v-align,center);
  padding: var(--card-header-padding,1rem);
}

<header
  class="c0"
/>
`;

exports[`<CardHeader /> should render bgc color orange 1`] = `
.c0 {
  background-color: var(--color-orange);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: var(--card-header-h-align,center);
  -webkit-justify-content: var(--card-header-h-align,center);
  -ms-flex-pack: var(--card-header-h-align,center);
  justify-content: var(--card-header-h-align,center);
  -webkit-align-items: var(--card-header-v-align,center);
  -webkit-box-align: var(--card-header-v-align,center);
  -ms-flex-align: var(--card-header-v-align,center);
  align-items: var(--card-header-v-align,center);
  padding: var(--card-header-padding,1rem);
}

<header
  class="c0"
/>
`;

exports[`<CardHeader /> should render bgc color purple 1`] = `
.c0 {
  background-color: var(--color-purple);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: var(--card-header-h-align,center);
  -webkit-justify-content: var(--card-header-h-align,center);
  -ms-flex-pack: var(--card-header-h-align,center);
  justify-content: var(--card-header-h-align,center);
  -webkit-align-items: var(--card-header-v-align,center);
  -webkit-box-align: var(--card-header-v-align,center);
  -ms-flex-align: var(--card-header-v-align,center);
  align-items: var(--card-header-v-align,center);
  padding: var(--card-header-padding,1rem);
}

<header
  class="c0"
/>
`;

exports[`<CardHeader /> should render bgc color red 1`] = `
.c0 {
  background-color: var(--color-red);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: var(--card-header-h-align,center);
  -webkit-justify-content: var(--card-header-h-align,center);
  -ms-flex-pack: var(--card-header-h-align,center);
  justify-content: var(--card-header-h-align,center);
  -webkit-align-items: var(--card-header-v-align,center);
  -webkit-box-align: var(--card-header-v-align,center);
  -ms-flex-align: var(--card-header-v-align,center);
  align-items: var(--card-header-v-align,center);
  padding: var(--card-header-padding,1rem);
}

<header
  class="c0"
/>
`;

exports[`<CardHeader /> should render bgc color yellow 1`] = `
.c0 {
  background-color: var(--color-yellow);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: var(--card-header-h-align,center);
  -webkit-justify-content: var(--card-header-h-align,center);
  -ms-flex-pack: var(--card-header-h-align,center);
  justify-content: var(--card-header-h-align,center);
  -webkit-align-items: var(--card-header-v-align,center);
  -webkit-box-align: var(--card-header-v-align,center);
  -ms-flex-align: var(--card-header-v-align,center);
  align-items: var(--card-header-v-align,center);
  padding: var(--card-header-padding,1rem);
}

<header
  class="c0"
/>
`;
